src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,1022 @@
+        /// <param name="xShutPin">
+        /// The xShut pin number used to power up the device.
+        /// If omitted, The xShutPin must be tied to the power supply value through a pull up resistor
+        /// </param>
+        /// <param name="operationTimeoutMilliseconds">Timeout for reading data, by default 500 milliseconds</param>
+        public Vl53L1X(byte i2cAddress = (byte)Registers.VL53L1X_DEFAULT_DEVICE_ADDRESS, int busId = 1, int xShutPin = -1, int operationTimeoutMilliseconds = 500)
+        {
+            _xShutPin = xShutPin;
+            _operationTimeoutMilliseconds = operationTimeoutMilliseconds;
+
+            PowerOn();
+            ChangeI2CAddress(i2cAddress, busId);
+            WaitForBooted();
+            InitSensor();
+        }
+
+        /// <summary>
+        ///     Get a distance in millimeters.
+        ///     If ranging has not been started yet, the function will automatically start the ranging feature of the device.
+        /// </summary>
+        public ushort Distance
+        {
+            get
+            {
+                if (!_rangingInitialized)
+                {
+                    StartRanging();
+                }
+
+                WaitForDataReady();
+                ushort distance = GetDistance();
+                return distance;
+            }
+        }
+
+        private void PowerOn()
+        {
+            if (_xShutPin < 0)
+            {
+                return;
+            }
+
+            _controller = new GpioController();
+            _controller.OpenPin(_xShutPin, PinMode.Output);
+            _controller.Write(_xShutPin, PinValue.Low);
+            Thread.Sleep(10);
+            _controller.Write(_xShutPin, PinValue.High);
+            Thread.Sleep(10);
+        }
+
+        private void PowerOff()
+        {
+            if (_xShutPin < 0 || _controller == null)
+            {
+                return;
+            }
+
+            _controller.Write(_xShutPin, PinValue.Low);
+            Thread.Sleep(10);
+            _controller.ClosePin(_xShutPin);
+            _controller.Dispose();
+            _controller = null;
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            _i2CDevice.Dispose();
+            PowerOff();
+        }
+
+        private void WaitForBooted()
+        {
+            BootState bootState = GetBootState();
+            Stopwatch watch = Stopwatch.StartNew();
+
+            while (bootState != BootState.Booted && watch.ElapsedMilliseconds < _operationTimeoutMilliseconds)
+            {
+                Thread.Sleep(10);
+                bootState = GetBootState();
+            }
+
+            if (bootState != BootState.Booted)
+            {
+                throw new IOException(
+                    $"The device did not boot withing the specified timeout of {_operationTimeoutMilliseconds} ms");
+
+            }
+        }
+
+        private void ChangeI2CAddress(byte newAddress, int busId)
```

Comment:
You will have to adjust this function to align with our conventions regarding the I2cDevice usage.

Reply:
The function is now the same as it is with it's sister device